lib/tasks/server.rake

```diff
@@ -18,17 +18,24 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 #++
 
-require_relative "../yast/tarball_server"
+begin
+  require_relative "../yast/tarball_server"
+rescue LoadError
+  puts "webrick is not available" if verbose == true
+end
 
-# Rake task for running a source code web server,
-# designed for the `yupdate` script.
-desc "Start an HTTP server providing dynamically generated source code tarball"
-task :server, [:port] do |_task, args|
-  server = Yast::TarballServer.new(args[:port])
+# load server only if webrick is available
+if defined?(Yast::TarballServer)
```

Comment:
I'd move the check inside the task, this solution has a disadvantage that the task will not be visible (in `rake -T`) and fail when you call it and you do not know why.

Additionally it should tell you how to fix that.

```
desc "Start ..."
task :server, ...
  begin
    require_relative "../yast/tarball_server"
  rescue LoadError
    abort("Webrick server is not installed, please install the webrick Ruby gem")
  end
```

Please also update the main `README.md` and document that with Ruby 3 the task needs the webrick gem.

Reply:
makes sense for me